From ea760fba4a2b23f75ffcca7b79ccb6333fcee5d9 Mon Sep 17 00:00:00 2001
From: Wenlin Kang <wenlin.kang@windriver.com>
Date: Wed, 19 Mar 2014 14:35:35 +0800
Subject: [PATCH] x86/perf: fixed a perf event issue

JIRAID: LIN5-18141

When enables LOCKUP_DETECTOR option, the cycle counter is no
longer available for perf tool. eg,for "perf stat" can only
display results as follows:
	...
 <not counted> cycles
	...

and for "perf top -e cycles" will fail completely.

Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/x86/kernel/cpu/perf_event_intel.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/x86/kernel/cpu/perf_event_intel.c b/arch/x86/kernel/cpu/perf_event_intel.c
index 3a447a1..d695418 100644
--- a/arch/x86/kernel/cpu/perf_event_intel.c
+++ b/arch/x86/kernel/cpu/perf_event_intel.c
@@ -2223,7 +2223,7 @@ __init int intel_pmu_init(void)
 		 * counter, so do not extend mask to generic counters
 		 */
 		for_each_event_constraint(c, x86_pmu.event_constraints) {
-			if (c->cmask != X86_RAW_EVENT_MASK
+			if (c->cmask != FIXED_EVENT_FLAGS
 			    || c->idxmsk64 == INTEL_PMC_MSK_FIXED_REF_CYCLES) {
 				continue;
 			}
-- 
1.7.5.4

